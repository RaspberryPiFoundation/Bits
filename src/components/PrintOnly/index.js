import { Base } from 'react-iotacss'
import classnames from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'

/**
 * Ensures contents are hidden from display, but included in print
 */
export const PrintOnly = ({ children, className, ...props }) => {
  const classNames = classnames(className, 'u-print-only')

  return (
    <Base className={classNames} {...props}>
      {children}
    </Base>
  )
}

PrintOnly.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
}

export default PrintOnly
