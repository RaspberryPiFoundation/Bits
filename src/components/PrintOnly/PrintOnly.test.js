import { shallow } from 'enzyme'
import PrintOnly from '../PrintOnly'
import React from 'react'

let printOnly

const children = 'Whatever'
const className = 'some-extra-class-name'

describe('<Slice />', () => {
  describe('Renders...', () => {
    beforeEach(() => {
      printOnly = shallow(<PrintOnly>{children}</PrintOnly>)
    })

    it('without crashing', () => {
      expect(printOnly.find('.u-print-only')).toBePresent()
    })

    it('matches its snapshot', () => {
      expect(printOnly).toMatchSnapshot()
    })
  })

  describe('With classNames', () => {
    it('adds the correct className', () => {
      printOnly = shallow(
        <PrintOnly className={className}>{children}</PrintOnly>,
      )
      expect(printOnly.find('.u-print-only')).toHaveClassName(className)
    })
  })
})
