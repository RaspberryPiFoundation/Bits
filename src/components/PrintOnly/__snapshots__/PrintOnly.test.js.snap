// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slice /> Renders... matches its snapshot 1`] = `
<Base
  className="u-print-only"
  tagName="div"
>
  Whatever
</Base>
`;
